packages/next/src/lib/waiter.test.ts

```diff
@@ -0,0 +1,46 @@
+import { Waiter } from './waiter'
+
+describe('Waiter', () => {
+  describe("it will time out if promise doesn't resolve in time", () => {
+    it.each([100, 50, 10])('for %dms', async (ms) => {
+      const waiter = new Waiter('Time out')
+
+      setTimeout(
+        () => {
+          waiter.done()
+        },
+        // We always finish 10ms after the waiter should wait for.
+        ms + 10
+      )
+
+      await expect(() => waiter.wait(ms)).rejects.toThrow()
```

Comment:
would it be worth verifying we throw the error message specified in the waiter constructor?